klient/decoder/decoder.go

```diff
@@ -92,6 +105,30 @@ func DecodeAllFiles(ctx context.Context, fsys fs.FS, pattern string, options ...
 	return objects, err
 }
 
+// ApplyUsingManifestDir resolves all the files in the Directory dirPath against the globbing pattern and creates a kubernetes
+// resource for each of the resources found under the manifest directory.
+func ApplyUsingManifestDir(ctx context.Context, r *resources.Resources, dirPath, pattern string, f PreHandlerFuncHook, createOptions []resources.CreateOption, options ...DecodeOption) error {
```

Comment:
Suggestion: `ApplyWithManifestDir` to stay consistent with naming.

Reply:
Done